src/lib/swapper/swappers/utils/helpers/helpers.ts

```diff
@@ -76,71 +58,42 @@ export const getERC20Allowance = ({
   return erc20Contract.methods.allowance(ownerAddress, spenderAddress).call()
 }
 
-export const isApprovalRequired = async ({
+export const getFeesFromContractData = async ({
+  accountNumber,
   adapter,
-  receiveAddress,
-  allowanceContract,
-  sellAsset,
-  sellAmountExcludeFeeCryptoBaseUnit,
-  web3,
-  erc20AllowanceAbi,
-}: IsApprovalRequiredArgs): Promise<boolean> => {
-  try {
-    if (sellAsset.assetId === adapter.getFeeAssetId()) {
-      return false
-    }
-
-    const ownerAddress = receiveAddress
-    const spenderAddress = allowanceContract
-
-    const { assetReference: sellAssetErc20Address } = fromAssetId(sellAsset.assetId)
-
-    const allowanceOnChain = await getERC20Allowance({
-      web3,
-      erc20AllowanceAbi,
-      ownerAddress,
-      spenderAddress,
-      sellAssetErc20Address,
-    })
-    if (!allowanceOnChain) {
-      throw new SwapError(`[isApprovalRequired] - No allowance data`, {
-        details: { allowanceContract, receiveAddress },
-        code: SwapErrorType.RESPONSE_ERROR,
-      })
-    }
-
-    if (bn(allowanceOnChain).isZero()) return true
-
-    const allowanceRequired = bnOrZero(sellAmountExcludeFeeCryptoBaseUnit).minus(allowanceOnChain)
-    return allowanceRequired.gt(0)
-  } catch (e) {
-    if (e instanceof SwapError) throw e
-    throw new SwapError('[isApprovalRequired]', {
-      cause: e,
-      code: SwapErrorType.ALLOWANCE_REQUIRED_FAILED,
-    })
-  }
-}
-
-export const getFeesFromFeeData = async ({
+  erc20ContractAddress,
+  erc20ContractData,
   wallet,
-  feeData,
-}: GetFeesFromFeeDataArgs): Promise<evm.Fees & { gasLimit: string }> => {
+}: GetFeesFromContractDataArgs): Promise<evm.Fees & { gasLimit: string }> => {
   if (!supportsETH(wallet)) {
-    throw new SwapError('[getFeesFromFeeData]', {
+    throw new SwapError('[getFeesFromContractData]', {
       cause: 'eth wallet required',
       code: SwapErrorType.SIGN_AND_BROADCAST_FAILED,
       details: { wallet },
     })
   }
 
-  const gasLimit = feeData.estimatedGasCryptoBaseUnit
-  const gasPrice = feeData.gasPriceCryptoBaseUnit
-  const maxFeePerGas = feeData.maxFeePerGas
-  const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas
+  const ownerAddress = await adapter.getAddress({ accountNumber, wallet })
```

Comment:
would just use `from`, along the same lines of gomes vernacular, easier to use the same language as jsonrpc rather than have different ways of naming things throughout the stack